modules/service/src/main/scala/ch/epfl/bluebrain/nexus/admin/service/encoders/RoutesEncoder.scala

```diff
@@ -15,9 +15,8 @@ import ch.epfl.bluebrain.nexus.commons.http.JsonOps._
 /**
   * Constructs implicit encoders used to format HTTP responses.
   * @param coreContext core context to inject into the response
-  * @param extractId   function which extracts Id from Entity
   */
-class RoutesEncoder[Entity](coreContext: ContextUri)(implicit extractId: (Entity) => Id) {
+class RoutesEncoder[Entity]()(implicit coreContext: ContextUri) {
```

Comment:
This can be a trait `trait RoutesEncoder[Entity]` and the `implicit coreContext: ContextUri` can be directly into the method, since it is not required for all the encoders.
In this way, you can mixing this trait in the `ProjectRoutes` --> `...extends BaseRoute with RoutesEncoder[Project]` and the imports will be directly available